python/docs/curves.rst

```diff
@@ -1,19 +1,103 @@
+.. currentmodule:: dlisio.plumbing
+
 Curves
-------
+======
+
+The primary data of a dlis-file are curves, also referred to as channels.
+Typically a curve is the recorded measurements taken along the borehole,
+indexed against an axis, like depth or time. But a curve can also be a
+computation of some sort, e.g. a calibrated version of a measured curve.
+
+In dlisio, curves are accessed through :class:`Frame`- or
+:class:`Channel`-objects, by calling their :func:`curves` methods.  The primary
+data type for curves is `structured numpy.ndarray`_.  This enables quick and
+easy mathematical operations on the data you care about.
+
+Frame-objects
+-------------
+
+Curves are organized in frames. Conceptually, a frame can be seen as a table of
+data where each column is a curve, like in the time-indexed frame below.
+Frames almost always have an index channel that provides the position in e.g.
+depth or time at which the rest of the values in the row were measured. Each
+frame usually corresponds to a log run, but otherwise frames impose little
+structure except grouping channels that have a common index. There is no
+restrictions on the number of channels per frame, or the number of frames in a
+file.
+
+Frames are described by :class:`Frame`-objects. These contain a list of
+:class:`Channel`-objects, which describes the individual curves in more detail.
+Additionally, the :class:`Frame` holds information about the index in which
+*all* the channels in the frame are index against. See :class:`Frame` for a
+full list it`s attributes.
+
+.. csv-table:: _`A short TIME indexed frame`
+   :header: "FRAMENO", "TIME", "TDEP", "ETIM", "LMVL", "UMVL", "CFLA", "OCD"
+   :widths: 10, 20, 20, 20, 20, 20, 20, 20
+
+    1 , 16677259 , 852606.0 , 0.  , 585 , 635 , 18 , 6789.05
+    2 , 16677659 , 852606.0 , 0.4 , 585 , 635 , 18 , 6789.05
+    3 , 16678059 , 852606.0 , 0.8 , 585 , 635 ,  0 , 6789.05
+    4 , 16678459 , 852606.0 , 1.2 , 585 , 635 ,  0 , 6789.05
+    5 , 16678859 , 852606.0 , 1.6 , 585 , 635 ,  0 , 6789.05
+
+.. note::
+   FRAMENO: The first column of every Frame is always FRAMENO, which is the row
+   numbers. These are recorded in the actual file. Generally FRAMENO is not
```

Comment:
2. maybe not, the intension was to underline that because they are in the file, they can be used for debugging wierd stuff. but its kinda implied anyways so ill remove it